ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/RelationEdition/RelationEditionModal.js

```diff
@@ -0,0 +1,179 @@
+import { DocumentEdition } from '@components/Document';
+import { HitsSearch } from '@components/ESSelector/HitsSearch';
+import { DocumentSelectListEntry } from '@pages/backoffice/components/Document';
+import SeriesSelectListEntry from '@pages/backoffice/components/Series/SeriesSelectListEntry/SeriesSelectListEntry';
+import { MultipleSelections } from '@pages/backoffice/Document/DocumentDetails/components/DocumentRelations/components/MultipleSelections';
+import { RelationModal } from '../components';
+import { RelationSummary } from '@pages/backoffice/Document/DocumentDetails/components/DocumentRelations/components/RelationSummary';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Container, Form, Icon, Label, Modal } from 'semantic-ui-react';
+import { document as documentApi, series as seriesApi } from '@api';
+import isEmpty from 'lodash/isEmpty';
+import remove from 'lodash/remove';
+
+export default class RelationEditionModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selections: [],
+      isLoading: false,
+      currentRecordType: 'document',
+      query: documentApi.list,
+      selectOpen: false,
+    };
+  }
+
+  removeFromSelection = (e, { dataPid }) => {
+    const { selections } = this.state;
+    remove(selections, o => o.metadata.pid === dataPid);
+
+    this.setState({ selections: selections });
+  };
+
+  onSelectResult = result => {
+    const { id } = result;
+    const { relations } = this.props;
+    const { selections } = this.state;
+
+    /* if relation already exists, nothing happens on select */
+    if (
+      isEmpty(relations.edition) ||
+      !relations.edition.find(o => o.pid === result.metadata.pid)
+    ) {
+      result.id = id;
+      result.key = id;
+      result.disabled = false;
+
+      selections.push(result);
+      this.setState({ selections: selections });
+    } else {
+      result.id = id;
+      result.key = id;
+      result.disabled = true;
+    }
+  };
+
+  selectResultRender = option => {
+    const existingRelations = this.props.relations;
+    const currentSelections = this.state.selections;
+
+    let disabled = false;
+    if (
+      (!isEmpty(existingRelations.edition) &&
```

Comment:
May I suggest to refactor it for readability? E.g.
```
const hasEdition = !isEmpty(existingRelations.edition);
const hasExistingRelation = existingRelations.edition.find(o => o.pid === option.metadata.pid));
...

const disabled = hasEdition && hasExistingRelation ...;
```

Reply:
👍